File: backend/models/Device.js
const mongoose = require('mongoose');

const deviceSchema = new mongoose.Schema({
  name: String,
  serial_number: { type: String, unique: true },
  cleaning_reagent: String,
  manufacturer: String,
  status: { type: String, enum: ['active','inactive','maintenance'], default: 'active' },
  staff_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Staff' },
  capacity_of_sample: Number,
  maintenance_schedule: { type: String, enum: ['daily','weekly','monthly'] },
  owner_id: { type: mongoose.Schema.Types.ObjectId, ref: 'LabOwner' }
});

module.exports = mongoose.model('Device', deviceSchema);



/*
{
    "success": true,
    "message": "✅ Registration submitted successfully! Please visit the lab for verification.",
    "registration": {
        "order_id": "691488ce1e87997695607712",
        "barcode": "ORD-1762953422785",
        "patient_name": "Ahmed Salem",
        "email": "ahmed.salem@example.com",
        "phone_number": "01123456789",
        "tests_ordered": [
            {
                "test_name": "Liver Function Test",
                "test_code": "LFT-005",
                "price": 220
            },
            {
                "test_name": "Hemoglobin A1c (HbA1c)",
                "test_code": "HBA1C-004",
                "price": 180
            },
            {
                "test_name": "Vitamin D (25-Hydroxyvitamin D)",
                "test_code": "VITD-007",
                "price": 280
            }
        ],
        "total_cost": 680,
        "tests_count": 3,
        "status": "pending",
        "next_steps": [
            "Visit the lab with your ID",
            "Show your registration barcode to staff",
            "Staff will verify your information and create your account",
            "You'll receive your account credentials via email"
        ]
    }
}   */

File: frontend_flutter/lib/models/device.dart
class Device {
  final String id;
  final String name;
  final String serialNumber;
  final String? manufacturer;
  final String? model;
  final DateTime? purchaseDate;
  final DateTime? warrantyExpiry;
  final String status; // 'active', 'inactive', 'maintenance'
  final String? staffId;
  final String ownerId;
  final DateTime? createdAt;

  Device({
    required this.id,
    required this.name,
    required this.serialNumber,
    this.manufacturer,
    this.model,
    this.purchaseDate,
    this.warrantyExpiry,
    this.status = 'active',
    this.staffId,
    required this.ownerId,
    this.createdAt,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      serialNumber: json['serial_number'] ?? '',
      manufacturer: json['manufacturer'],
      model: json['model'],
      purchaseDate: json['purchase_date'] != null
          ? DateTime.parse(json['purchase_date'])
          : null,
      warrantyExpiry: json['warranty_expiry'] != null
          ? DateTime.parse(json['warranty_expiry'])
          : null,
      status: json['status'] ?? 'active',
      staffId: json['staff_id'] is Map
          ? json['staff_id']['_id']
          : json['staff_id'],
      ownerId: json['owner_id'] is Map
          ? json['owner_id']['_id']
          : json['owner_id'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'serial_number': serialNumber,
      if (manufacturer != null) 'manufacturer': manufacturer,
      if (model != null) 'model': model,
      if (purchaseDate != null)
        'purchase_date': purchaseDate!.toIso8601String(),
      if (warrantyExpiry != null)
        'warranty_expiry': warrantyExpiry!.toIso8601String(),
      'status': status,
      if (staffId != null) 'staff_id': staffId,
      'owner_id': ownerId,
      if (createdAt != null) 'createdAt': createdAt!.toIso8601String(),
    };
  }

  bool get isActive => status == 'active';
  bool get needsMaintenance => status == 'maintenance';
}
